cirq/contrib/qcircuit/qcircuit_diagrammable.py

```diff
@@ -46,36 +47,63 @@ def __init__(self, *symbols: str) -> None:
         self.symbols = symbols
 
     def qcircuit_diagram_info(self, args: ops.TextDiagramInfoArgs
-                              ) -> Tuple[str, ...]:
-        return ops.TextDiagramInfo(self.symbols).wire_symbols
-
+                              ) -> ops.TextDiagramInfo:
+        return ops.TextDiagramInfo(self.symbols)
+
+def _get_multigate_parameters(gate: ops.TextDiagrammable,
+                              args: ops.TextDiagramInfoArgs
+                              ) -> Optional[Tuple[int, int]]:
+    if not isinstance(gate, gate_features.InterchangeableQubitsGate):
+        return None
+    if args.qubit_map is None or args.known_qubits is None:
+        return None
+
+    indices = [args.qubit_map[q] for q in args.known_qubits]
+    min_index = min(indices)
+    n_qubits = len(args.known_qubits)
+    if sorted(indices) != list(range(min_index, min_index + n_qubits)):
+        return None
+    return min_index, n_qubits
 
 class _TextToQCircuitDiagrammable(QCircuitDiagrammable):
     def __init__(self, sub: ops.TextDiagrammable) -> None:
         self.sub = sub
 
     def qcircuit_diagram_info(self, args: ops.TextDiagramInfoArgs
-                              ) -> Tuple[str, ...]:
+                              ) -> ops.TextDiagramInfo:
         info = self.sub.text_diagram_info(args)
-
+        multigate_parameters = _get_multigate_parameters(self.sub, args)
+        if multigate_parameters is not None:
```

Comment:
Put this `if` block at the end of `_get_multigate_parameters` and have that method return `Optional[ops.TextDiagramInfo]`. That will likely fix the type error.